### ISSUE TITLE
Optional chaining operator on props produces wrong output in angular

### ISSUE THREAD

**Issue by chadalen** [BuilderIO/mitosis#886]
### I am interested in helping provide a fix!

Yes

### Which generators are impacted?

- [ ] All
- [X] Angular
- [ ] HTML
- [ ] Qwik
- [ ] React
- [ ] React-Native
- [ ] Solid
- [ ] Stencil
- [ ] Svelte
- [ ] Vue
- [ ] Web components

### Reproduction case

https://mitosis.builder.io/?outputTab=IYOw5grgNsBOQ%3D%3D%3D&code=JYWwDg9gTgLgBAbzgVwM4FMDKMCGN1wC%2BcAZlBCHAEQACARssADYAm6UAdMBAPQjAwIqYKioBuAFAT0AD0iw4bEjmRN4JZADsAxjG6a4AWQCeAYQqRN6TTAAUYcmFQBKRBLhwo6GMigHb7h5wADwswABuAHyBQSFh4TGxcMAsALwIDhBOAPxcLISJsdpMOKioAHI4IOjpmTkcqBTopiVlBUlBhXDRHR4sEOiocJoQ8ADu0ADWXcE88T1Bs%2FOBzpKEQA%3D

### Expected Behaviour

```
export default function MyComponent(props) {
  const state = useStore({
    value: true,
    visibile: false,
  });

  const someRef = useRef(null);

  return (
    <div
      ref={someRef}
      className={classNames({ 'some-class': state.value })}
    >
      testing
    </div>
  );
}
```

should produce

```
@Component({
  selector: "my-component, MyComponent",
  template: `
    <div>
      <div [attr.id]="id" [class]="someClass">does not work</div>
    </div>
  `,
})
export class MyComponent {
  @Input() id: any;
  @Input() someClass: any;
}
```

### Actual Behaviour

This
```
export default function MyComponent(props) {
  return (
    <div>
      <div
        id={props?.id}
        className={props?.someClass}
        
      >
        does not work
      </div>
    </div>
  );
}
```

produces

```
    <div>
      <div [attr.id]="props?.id" [class]="props?.someClass">does not work</div>
    </div>
```

but `props` is not defined anywhere.

**Comment by samijaber** [comment]
But a `props` object should always exist, so it never makes sense to use the optional chaining operator to access one of its properties. It should always be `props.id`, never `props?.id`.

Can you explain why you're using optional chaining here?

**Comment by chadalen** [comment]
> 

You are right however... In React you CAN do this which works. So this would prevent other react devs running into issues like this while working in mitosis.

I would suggest updating the linter or converting it if someone were to write code like this so the output "just works".

**Comment by samijaber** [comment]
That's fair. Was mostly pointing out that it's never technically needed in the Mitosis code.

Fixing this might be as simple as updating this regex check:

https://github.com/BuilderIO/mitosis/blob/a762948e6014450d5b8b80c961b771df4568f847/packages/core/src/helpers/strip-state-and-props-refs.ts#L67-L71

to include an optional `?`, e.g. `/props(\??)\./`

**Comment by decadef20** [comment]
> That's fair. Was mostly pointing out that it's never technically needed in the Mitosis code.
> 
> Fixing this should be as simple as updating this regex check:
> 
> https://github.com/BuilderIO/mitosis/blob/a762948e6014450d5b8b80c961b771df4568f847/packages/core/src/helpers/strip-state-and-props-refs.ts#L67-L71
> 
> to include an optional `?`, e.g. `/props(\??)\./`

I have tried this, but it doesn't work. It's more complicated than expected.  Eslint plugin may be a better way to handle it if need
